### PR TITLE
DCOS-10987: [2/2] Separate JSON Flow

### PR DESCRIPTION
**Pull request by wavesoft**
---
⚠️  Depends on https://github.com/dcos/dcos-ui/pull/1408

---

This PR introduces the first draft of the json-only editor view. There is proper error validation and integration with the review screen.

new-json-only mov

The following things are missing:

- [x] Use the correct classes to make the JSON editor occupy the full size -- @jfurrow I might need some help here
- [x] Fix the floating error message (caused by the hack required on the side-by-side view)
- [x] Fix an annoying bug that displays an error the first time the JSON editor is shown

### CODE REVIEW

**Comment by jfurrow** [comment]
@wavesoft do you know if theres a reason we can't update `react-ace` to the most recent version? I tested this quickly and am receiving an error `Uncaught TypeError: Super expression must either be null or a function, not undefined` which seems to come from the `JobFormModal`. I did not investigate any further than this, sorry!

The new version allows us to pass custom styles into the editor itself, so the attribute `style={{height: '100%', width: '100%', position: 'absolute'}}` on `JSONEditor` in `CreateServiceJsonOnly.js` would do the trick!

If there is a problem with upgrading `react-ace`, then we will have to override some styles with the `!important`. So if you go this route, just add these styles to `components/modal/full-screen` and then pass `className="modal-full-screen-fill-body"` to `JSONEditor`:
```less
.modal-full-screen-fill-body {
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;

    &.ace_editor {
      // Override inline styles on the Ace editor.
      height: 100% !important;
      position: absolute !important;
    }
  }
```

Feel free to ping me if you'd like any additional help with this!

**Comment by wavesoft** [comment]
Hey @jfurrow . Thanks for the tips. This version also supports providing `width` and `height`, check the `JSONEditor` component. The problem with the latest version is that it also needs the latest `React`. Apparently they have changed something on their API.

I will try your suggestions and check how it goes.

**Comment by wavesoft** [comment]
Btw, I was referring to the `4.x` branch. I actually updated to the latest `3.x` branch (that still uses the old React API) and it works fine.

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "prettycron": "0.10.0",
     "query-string": "4.1.0",
     "react": "0.14.7",
-    "react-ace": "3.4.1",
+    "react-ace": "3.7.0",
```
Comment (mlunoe) [inline package.json]: Have you checked the implications of this update with the old flow?
Does that still work?

Comment (wavesoft) [inline package.json]: Yes, there is a big change on the `4.x` branch. `3.x` is still compatible like before. In addition we now have a single place the ACE editor is used (the `JSONEditor` component) so if something goes wrong, that's the only place we have to touch.

**Comment by orlandohohmeier** [comment]
validation-issue

For some reason the validation isn't working when I paste a app/pod definition.

**Comment by wavesoft** [comment]
Everything should be addressed 👍

**Pull request merged**